WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -99,31 +107,43 @@ final class ProductDetailPreviewViewModel: ObservableObject {
                                                     tone: aiTone)
             generatedProduct = product
             isGeneratingDetails = false
+            shouldShowFeedbackView = true
+            analytics.track(event: .ProductCreationAI.generateProductDetailsSuccess())
         } catch {
+            analytics.track(event: .ProductCreationAI.generateProductDetailsFailed(error: error))
             DDLogError("⛔️ Error generating product with AI: \(error)")
             errorState = .generatingProduct
         }
     }
 
     @MainActor
     func saveProductAsDraft() async {
+        analytics.track(event: .ProductCreationAI.saveAsDraftButtonTapped())
         guard let generatedProduct else {
             return
         }
         errorState = .none
         isSavingProduct = true
         do {
             let newProduct = try await saveProductRemotely(product: generatedProduct)
+            analytics.track(event: .ProductCreationAI.saveAsDraftSuccess())
             onProductCreated(newProduct)
         } catch {
             DDLogError("⛔️ Error saving product with AI: \(error)")
+            analytics.track(event: .ProductCreationAI.saveAsDraftFailed(error: error))
             errorState = .savingProduct
         }
         isSavingProduct = false
     }
 
     func handleFeedback(_ vote: FeedbackView.Vote) {
-        // TODO
+        analytics.track(event: .AIFeedback.feedbackSent(source: .productCreation,
+                                                        isUseful: vote == .up))
+
+        // Delay the disappearance of the banner for a better UX.
+        DispatchQueue.main.asyncAfter(deadline: .now() + delayBeforeDismissingFeedbackBanner) { [weak self] in
+            self?.shouldShowFeedbackView = false
+        }
```

Comment:
💡  In SwiftUI we may not need this delay. Instead, try calling `withAnimation { viewModel.handleFeedback }` and in this method we toggle `shouldShowFeedbackView`. That way we can avoid the complication of `DispatchQueue`.

Reply:
Thanks for the tip. Changed this in 7c0fbe51c289a587aebf9422c586aab1a5c142a6